src/irmin-pack/traverse_pack_file.ml

```diff
@@ -56,7 +61,65 @@ end = struct
      as requests for more data. *)
   exception Not_enough_buffer
 
-  type index_binding = { key : Hash.t; data : Index.value }
+  type index_value = int63 * int * Pack_value.Kind.t
+  [@@deriving irmin ~equal ~pp]
+
+  type index_binding = { key : Hash.t; data : index_value }
+
+  module Index_reconstructor = struct
+    let create ?output config =
+      let dest =
+        match output with
+        | Some path -> path
```

Comment:
New weird edge case: `path` is explicitly passed and is equal to `Conf.root config`, but `Conf.readonly config`. Perhaps `failwith` if the supplied path already `IO.exist`s on disk?

Reply:
Good idea!